### PR TITLE
Add trinary and input validators.

### PR DESCRIPTION
**Pull request by oopsmonk**
Added:
* is_trytes
* is_trits
* is_tryte_with_trailing_zero
* is_address
* is_address_with_checksum
* is_tag
* is_seed
* is_security_level

### CODE REVIEW

File: utils/input_validators.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ */
+bool is_seed(tryte_t const* const seed);
+
+/**
+ * @brief Checks if the given number is a security level.
+ *
+ * @param[in] level An integer.
+ * @return boolean
+ */
+bool is_security_level(int level);
```
Comment (thibault-martinez) [inline utils/input_validators.h]: u8 ?

File: common/trinary/tryte.c [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+#include <string.h>
+
+#include "common/trinary/tryte.h"
+
+bool is_trytes(tryte_t const *const trytes, size_t len) {
+  size_t trytes_cnt = strlen((char *)trytes);
+  if (trytes_cnt != len) {
+    // length is not matched.
```
Comment (thibault-martinez) [inline common/trinary/tryte.c]:
```suggestion
    // length doesn't match.
```

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: utils/input_validators.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    return is_tryte_with_trailing_zero(address[NUM_TRYTES_ADDRESS - 1]);
+  }
+  return false;
+}
+
+bool is_address_with_checksum(tryte_t const *const address) {
+  if (strlen((char *)address) != NUM_TRYTES_ADDRESS + 9) {
+    return false;
+  }
+  return is_tryte_with_trailing_zero(address[NUM_TRYTES_ADDRESS - 1]);
```
Comment (thibault-martinez) [inline utils/input_validators.c]: Just double-checking, is it on purpose that `is_trytes` is not called on this function ? Do we assume we already called `is_address` before ?

Comment (oopsmonk) [reply utils/input_validators.c]: No, is_trytes should be called here.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**